Controllers/NoteController.cs

```diff
@@ -0,0 +1,55 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Taskmanager.Controllers.InputModels;
+using Taskmanager.Controllers.ViewModels;
+using Taskmanager.Data.Entities;
+using Taskmanager.Services.Interfaces;
+
+namespace Taskmanager.Controllers
+{
+    [ApiController]
+    [Route("api/notes")]
+    public class NoteController : ControllerBase
+    {
+        private readonly INoteService _noteService;
+
+        public NoteController(INoteService noteService)
+        {
+            _noteService = noteService;
+        }
+
+        [HttpGet("{todoitemId}")]
+        public async Task<ActionResult<NoteViewModel>> GetByIdAsync([FromRoute] int todoitemId)
+        {
+            Note requiredNote = await _noteService.GetOneAsync(todoitemId);
+
+            return Ok(NoteViewModel.MapNote(requiredNote));
+        }
+
+        [HttpPost("{todoitemId}")]
+        public async Task<ActionResult> AddAsync ([FromRoute] int todoitemId, [FromBody] NoteInputModel inputModel)
+        {
+            Note note = NoteInputModel.MapNote(inputModel);
+
+            await _noteService.AddAsync(todoitemId, note);
```

Comment:
I left a comment that AddAsync is not necessary, read about difference between Add and AddAsync and use-cases for Async